### ISSUE TITLE
Non-function properties incorrectly present in type of classes' prototypes

### ISSUE THREAD

**Issue by timruffles** [Microsoft/TypeScript#11558]
**TypeScript Version:**  2.0.0

**Code**

``` ts
class A {
    foo = "hi";
}

class B {}

// this is fine
const t: { prototype: { foo: string } } = A;

// this complains "types of property 'prototype' are incompatible"
const u: { prototype: { foo: string } } = B;
```

**Expected behavior:**

Only function properties should exist in prototypes of classes. This reflects the runtime behaviour (all properties are assigned in constructor).

**Actual behavior:** 

Non-function properties appear in the types of classes' prototypes.

**Comment by RyanCavanaugh** [comment]
We found that a not-insignificant of people used `prototype` to create property default values. Since there's not any other way to declare that property and it's a very difficult property to use incorrectly, it was better (and simpler) overall to include instance properties in the prototype.

**Comment by timruffles** [comment]
Is this documented?

Quite surprising (even if you know about the ES6 pt prop restriction + transpiler's strategy).
